apps/web/pages/[user]/[type].tsx

```diff
@@ -14,13 +17,38 @@ import { ssrInit } from "@server/lib/ssr";
 export type AvailabilityPageProps = inferSSRProps<typeof getServerSideProps>;
 
 export default function Type(props: AvailabilityPageProps) {
-  return <AvailabilityPage {...props} />;
+  const { t } = useLocale();
+  return props.isDynamicGroupBooking && !props.profile.allowDynamicBooking ? (
+    <div className="h-screen dark:bg-neutral-900">
+      <main className="mx-auto max-w-3xl px-4 py-24">
+        <div className="space-y-6" data-testid="event-types">
+          <div className="overflow-hidden rounded-sm border dark:border-gray-900">
+            <div className="p-8 text-center text-gray-400 dark:text-white">
+              <h2 className="font-cal mb-2 text-3xl text-gray-600 dark:text-white">
+                {" " + t("unavailable")}
+              </h2>
+              <p className="mx-auto max-w-md">{t("user_dynamic_booking_disabled")}</p>
+            </div>
+          </div>
+        </div>
+      </main>
+    </div>
+  ) : (
+    <AvailabilityPage {...props} />
```

Comment:
I got `props: any` here

Reply:
That's not right. I don't get `props: any` here 🤔. They're anyway inferred from SSR. No change was made to this bit, except it is called conditionally now (when it is just one user or all the users in the dynamic group have allowed dynamic group booking).